### ISSUE TITLE
Relations to entities with composite keys

### ISSUE THREAD

**Issue by aldahick** [typeorm/typeorm#476]
Currently, the code for handling foreign keys only allows the referenced entity to have a single primary column ([as seen here](https://github.com/typeorm/typeorm/blob/master/src/metadata/JoinColumnMetadata.ts#L81)). Example code below:

```typescript
// Parent.ts
@Entity()
export class Parent {
  @PrimaryColumn()
  public id1: string;

  @PrimaryColumn()
  public id2: string;

  @OneToMany(t => Child, c => c.parent)
  public children: Child[];
}

// Child.ts
@Entity()
export class Child {
  @PrimaryGeneratedColumn()
  public id: number;

  @ManyToOne(t => Parent, p => children)
  public parent: Parent;
}
```

Are there plans to implement this functionality? Would a PR implementing this functionality be welcome?

**Comment by pleerock** [comment]
> Would a PR implementing this functionality be welcome?

for sure, but this is very complex to implement.

this is already implemented in #337 and part of big changes. This wont be shipped soon because there are other changes pending before new release with all those big changes (aprox. milestone 2-3 months)

**Comment by pleerock** [comment]
relations with composite keys is quite good now and its already released in latest 0.1.0-alpha versions
